lint-rules.md

```diff
@@ -0,0 +1,1280 @@
+# WDL Lint Rules
```

Comment:
## Overall Remarks

- (**Important**) Summarize the list of lints with "one-liner" summaries for someone to check over at a glance, with more details in the full docs linked to from the summaries. This will help get feedback from more people, and I'd be happy to broadcast a bit more but can't imagine many folks would want to spend 30-60 minutes reading through the entire doc. They might spend 10 minutes however.
- Include in the doc a method for submitting feedback, whether it be via issues, commenting on this PR, etc. I think issues make more sense in the future since this PR is closed and can get quickly cluttered in the future.
- "Groups" for the lint rules could act like tags: you can be a part of multiple groups, e.g. `import_sort` is part of `sorting` and `import` ? Though I'm not sure you'd ever want to turn off all `import` rules for some reason. But it's worth considering at this stage if you ever foresee potential overlaps.
- Maybe declare at the top that "a blank line" means "exactly one blank line" throughout the doc so it's clear everywhere it doesn't mean "at least one blank line."
- It might also make sense to reorganize the doc to have lint group subheaders? Maybe someone knows they have strong feelings about whitespace but don't care about deprecation lints and can just jump to that section for example when reviewing. It can also make it easier for folks to debug which lints they want for their project if they're having issues with a particular flavor of warnings.

Reply:
> *  Summarize the list of lints with "one-liner" summaries for someone to check over at a glance, with more details in the full docs linked to from the summaries.

Great idea! Will put this summary on my TODO list :)

> * Include in the doc a method for submitting feedback

Good point, this was an oversight not including our "preferred method of feedback". Will converse with the team and decide what would be best for us to track and respond to.

> * "Groups" for the lint rules could act like tags

Good idea to consider. The term "tags" is maybe more quickly understood than "groups"; so even if we don't want to start assigning multiple to a rule (now) it could make the intent clearer.

> * Maybe declare at the top that "a blank line" means "exactly one blank line" throughout the doc

👍 

> * It might also make sense to reorganize the doc to have lint group subheaders?

Another good suggestion! I think this would be a pretty straight-forward improvement.

We really appreciate your input.